File: week-01/reverse-string/reverse_string_test.go
package reversestring

import "testing"

func TestReverseString(t *testing.T) {
	tests := []struct {
		name string
		in   string
		want string
	}{
		{
			name: "single word",
			in:   "Cat",
			want: "taC",
		},
		{
			name: "single sentence",
			in:   "The Daily Byte",
			want: "etyB yliaD ehT",
		},
		{
			name: "palindrome",
			in:   "civic",
			want: "civic",
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := Reverse(tt.in); got != tt.want {
				t.Errorf("Reverse(%s) = %s, want %s", tt.in, got, tt.want)
			}
		})
	}
}

File: week-01/reverse-string/reverse_string_test.py
from reverse_string import reverse


def test_reverse():
    tests = [
        {
            "input": "",
            "want": "",
        },
        {
            "input": "coding",
            "want": "gnidoc",
        },
        {
            "input": "level",
            "want": "level",
        },
    ]

    for test in tests:
        assert reverse(test["input"]) == test["want"]
